src/Foundation/NSObject2.cs

```diff
@@ -103,6 +105,19 @@ enum Flags : byte {
 			get { return ((flags & Flags.InFinalizerQueue) == Flags.InFinalizerQueue); }
 		}
 
+		bool IsCustomType {
+			get {
+				bool value;
+				var value = (flags & Flags.IsCustomType) == Flags.IsCustomType;
+				if (!value) {
```

Comment:
that's only faster for custom types
i.e. it still calls the (costly?) method for non-custom types
not sure if this matters (in other cases than the one provided)

Reply:
If `IsCustomType` returns false, `IsRegisteredToggleRef` will be set to true, and the next time `MarkDirty` is called on the same object it'll exit immediately.

This obviously only matters if `MarkDirty` is called multiple times per instance, which is exactly what's happening.